src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerAdditionService.java

```diff
@@ -0,0 +1,108 @@
+
+        var individualManagingOfficerAddition =
+                new IndividualManagingOfficerAddition(
+                        actionDate,
+                        addressDtoToAddress(residentialAddress),
+                        addressDtoToAddress(serviceAddress),
+                        resignedOn);
+
+        individualManagingOfficerAddition.setPersonName(new PersonName(mo.getFirstName(), mo.getLastName()));
+        individualManagingOfficerAddition.setFormerNames(mo.getFormerNames());
+        individualManagingOfficerAddition.setBirthDate(mo.getDateOfBirth());
+        individualManagingOfficerAddition.setRole(mo.getRoleAndResponsibilities());
+        individualManagingOfficerAddition.setOccupation(mo.getOccupation());
+
+        if (Objects.isNull(mo.getSecondNationality())) {
+            individualManagingOfficerAddition.setNationalityOther(mo.getNationality());
+        } else {
+            individualManagingOfficerAddition.setNationalityOther(
+                    String.format("%s, %s", mo.getNationality(), mo.getSecondNationality()));
+        }
+
+        return individualManagingOfficerAddition;
+    }
+
+    private void addCorporateManagingOfficerAdditions(
+            List<Addition> additions, OverseasEntitySubmissionDto overseasEntitySubmissionDto) {
+        var managingOfficerCorporate = overseasEntitySubmissionDto.getManagingOfficersCorporate();
+        if (managingOfficerCorporate != null) {
+            managingOfficerCorporate.stream()
+                    .filter(managingOfficer -> (managingOfficer.getChipsReference() == null))
+                    .map(this::getCorporateManagingOfficerAddition)
+                    .forEachOrdered(additions::add);
+        }
+    }
+
+    private CorporateManagingOfficerAddition getCorporateManagingOfficerAddition(ManagingOfficerCorporateDto mo) {
+        var actionDate = mo.getStartDate();
+        var residentialAddress = mo.getPrincipalAddress();
+        var serviceAddress = mo.getServiceAddress();
+        var resignedOn = mo.getResignedOn();
+
+        var corporateManagingOfficerAddition =
+                new CorporateManagingOfficerAddition(
+                        actionDate,
+                        addressDtoToAddress(residentialAddress),
+                        addressDtoToAddress(serviceAddress),
+                        resignedOn);
+
+        corporateManagingOfficerAddition.setName(mo.getName());
+        corporateManagingOfficerAddition.setContactName(mo.getContactFullName());
+        corporateManagingOfficerAddition.setContactEmail(mo.getContactEmail());
+
+        var legalForm = mo.getLegalForm();
+        var governingLaw = mo.getLawGoverned();
+        var registerLocation = mo.getPublicRegisterName();
+        var registrationNumber = mo.getRegistrationNumber();
+        var identification = new CompanyIdentification(
+                legalForm, governingLaw, registerLocation, null, registrationNumber);
```

Comment:
seems strange that the null is second last. Maybe registerLocation and registrationNumber need to be beside each other in construct arg list? Would need refactoring of other usages.